src/hooks/series/useSeriesData.ts

```diff
@@ -1,33 +1,33 @@
 import usePlaylist from '#src/hooks/usePlaylist';
-import useSeries from '#src/hooks/useSeries';
-import type { Playlist } from '#types/playlist';
+import useSeries from '#src/hooks/series/useSeries';
+import type { Playlist, PlaylistItem } from '#types/playlist';
 import type { Series } from '#types/series';
 
 const DEFAULT_DATA = { title: '', playlist: [] };
 
 type Data = {
-  seriesPlaylist: Playlist;
-  series: Series | undefined;
+  playlist: Playlist;
+  newSeries: { series: Series | undefined; media: PlaylistItem | undefined };
 };
 
 export const useSeriesData = (
-  entityId?: string,
+  legacySeriesPlaylistId: string | undefined,
+  seriesId: string | undefined,
 ): {
   data: Data;
-  // isItemError: boolean;
   isPlaylistError: boolean;
   isLoading: boolean;
 } => {
-  const { data: seriesData, isLoading: isSeriesLoading, error: seriesError, isError: isSeriesError } = useSeries(entityId);
+  const { data: seriesData, isLoading: isSeriesLoading, error: seriesError, isError: isSeriesError } = useSeries(seriesId);
```

Comment:
I think maybe it's time to get rid of 1 of the 2 useSeries hooks (useSeries and useSeriesData.) There are some complexities coming from the Promise.all and I just realized that you probably want to retrieve the media item for both the old and the new flows anyhow. Lastly, if we agree to drop episodes out of here and just use useEpisodes on a per page basis, this code becomes simple enough that we can eliminate the mental burden of what the difference between the useSeries and useSeriesData hooks is.

Reply:
It looks like we will still have to save the query to avoid unnecessary requests. But we can certainly combine them in one file.